Wbooks/src/redux/book/reducer.js

```diff
@@ -1,61 +1,26 @@
-import { actionTypes } from './actions';
+import { createReducer, completeReducer, completeState } from 'redux-recompose';
 
-const initialState = {
+import { actions } from './actions';
+
+const stateDescription = {
   bookDetail: null,
   bookList: [],
   wishList: [],
   rentedBooks: []
 };
 
-function reducer(state = initialState, action) {
-  switch (action.type) {
-    case actionTypes.LOAD_BOOK_DETAILS:
-      return {
-        ...state,
-        bookDetail: action.payload.book
-      };
-    case actionTypes.ADD_TO_WISHLIST:
-      return {
-        ...state,
-        wishList: [...state.wishList, action.payload.book]
-      };
-    case actionTypes.GET_RENTED_BOOKS:
-      return {
-        ...state
-      };
-    case actionTypes.GET_RENTED_BOOKS_SUCCESS:
-      return {
-        ...state,
-        rentedBooks: action.payload
-      };
-    case actionTypes.GET_RENTED_BOOKS_ERROR:
-      return {
-        ...state,
-        rentedBooks: []
-      };
-    case actionTypes.GET_BOOK_LIST_SUCCESS:
-      return {
-        ...state,
-        bookList: action.payload
-      };
-    case actionTypes.GET_BOOK_LIST_ERROR:
-      return {
-        ...state,
-        bookList: []
-      };
-    case actionTypes.RENT_BOOK:
-      return {
-        ...state,
-        rentedBooks: [...state.rentedBooks, action.payload.book]
-      };
-    case actionTypes.RETURN_BOOK:
-      return {
-        ...state,
-        rentedBooks: state.rentedBooks.filter(({ id }) => action.payload.book.id !== id)
-      };
-    default:
-      return state;
-  }
-}
+const initialState = completeState(stateDescription);
+
+const reducerDescription = {
+  primaryActions: [
+    actions.GET_BOOK_LIST,
+    actions.GET_RENTED_BOOKS,
+    actions.LOAD_BOOK_DETAILS,
+    actions.RETURN_BOOK,
```

Comment:
i'd use this effect `https://github.com/Wolox/redux-recompose/tree/master/src/effects/onDeleteByIndex`